Formula/harmony-one.rb

```diff
@@ -0,0 +1,113 @@
+class HarmonyOne < Formula
+  desc "Harmony Blockchain Node"
+  homepage "https://github.com/harmony-one/harmony"
+  # CONTRIBUTORS: Update the following (url, sha256)
+  url "https://github.com/harmony-one/harmony/archive/refs/tags/v4.3.0.tar.gz"
+  sha256 "d463cd60973509f2a7e906582a09a7cd21cf3638b956f164c63e0536a2063367"
+  license "LGPL-3.0-or-later"
+
+  # CONTRIBUTORS: Versions may need to be updated according to documentation at:
+  #    -  https://github.com/harmony-one/harmony
+
+  depends_on "bash"
+  depends_on "gmp"
+  depends_on "go@1.16"
+  depends_on "jq"
+  depends_on macos: :high_sierra
+  depends_on "make"
+  depends_on "openssl@1.1"
+  
+
+  # CONTRIBUTORS: Keep resources updated according to the changelog release tag:
+  #     - https://github.com/harmony-one/harmony/releases/tag/v#{version}
+
+  resource "bls" do
+    url "https://github.com/harmony-one/bls.git",
+        revision: "2b7e49894c0f15f5c40cf74046505b7f74946e52"
+  end
+
+  resource "mcl" do
+    url "https://github.com/harmony-one/mcl.git",
+        revision: "99e9aa76e84415e753956c618cbc662b2f373df1"
+  end
+
+  def install
+    # See https://github.com/golang/go/issues/26487
+    ENV.O0 if OS.linux?
+
+    # Setup Build Paths
+    ENV["GOPATH"] = buildpath
+
+    harmony_path = buildpath/"src/github.com/harmony-one/harmony"
+    harmony_path.install buildpath.children
+
+    %w[mcl bls].each do |r|
+      temp_dir = buildpath/"src/github.com/harmony-one/#{r}"
+
+      resource(r).stage do
+        mv pwd, temp_dir
+      end
+    end
+
+    cd buildpath/"src/github.com/harmony-one/harmony"
```

Comment:
Why not modern go modules?

Reply:
what do you mean by modern go modules? Not a go programmer but did research go modules. Don't understand what you are referring to. Not exactly a ruby wiz either. From my understanding this was an easy fix that required much less work than fixing errors during the build. Building the binary with the makefile requires the projects workdir to be setup in a specific manner as the other resources being build are dependencies. This solution was a simple way of correcting where homebrew was trying to run from and placing resources internally. Still dont see how go mod comes into play when it isnt used in the readme.md for this project in anyway other than how i used it?

Reply:
This directory structure isn't standard in modern go projects that use modules. And since modules have been required in Go for a while this is also an indication that the project isn't maintained that well and is likely to break often.

I'd see this as breaking: https://docs.brew.sh/Acceptable-Formulae#stuff-that-requires-heavy-manual-prepost-install-intervention

Reply:
How does that even relate to this ? It requires no manual post install intervention????

Reply:
It literally says that it needs to be in a resource block with a checksum wich it IS. Whats the deal?????

Reply:
> So that means the upstream tag is not building by default, that needs to be corrected upstream. With `go mod tidy` we're ignoring al the security measures in go modules and just downloading whatever latest version, which means it is not conforming to https://docs.brew.sh/Acceptable-Formulae#we-dont-like-install-scripts-that-download-unversioned-things and therefore not acceptable in homebrew-core.

How is it insecure to download the latest versions lmfao. It seems insecure to continually used outdated software that may have been patched due to a security vulnerability but what do i know im just a cyber security bootcamp instructor

Reply:
There's a whole lot more to security than just downloading the latest version of things. There's also
- source integrity verification: you get less of that with `go mod tidy` since you don't have the benefit of the sources being checksummed in the formula
- build reproducibility: see https://reproducible-builds.org